### PR TITLE
#16, #17, #19 and #20 - Remove portals, use native element constructors, support custom events and upgrade to use semantic-release / commitizen.

### PR DESCRIPTION
**Pull request by treshugart**
Implements #16, #17, #19 and #20. Bulk of the review `src/index.js`. It was simpler to do all of these at once since it was a major overhaul to the only file in this repo. The rest of the files were changed / added / removed from re-initialising from the updated build.

This PR also removes the old tests and adds new ones in their place as the old tests were tested against Skate and the portals method. Since we've generalised the wrapper and removed portal support, these tests were no longer relevant.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) {year} {author}
+Copyright (c) Trey Shugart <treshugart@gmail.com> 2016
```
Comment (joscha) [inline LICENSE]: This is flipped now, probably won't matter though :)

**Comment by joscha** [comment]
LGTM, however I think it would be beneficial to read the config files (webpack, karma, saucelabs) from skatejs-build like we do in named-slots if possible.

**Comment by treshugart** [comment]
Re the config files, I agree and that's been done. Originally I wanted to do this but it got lost in the midst of doing other fixes. Waiting for the build to re-release now.

File: LICENSE (hunk 1 above)
Comment (treshugart) [inline LICENSE]: Interesting. That's a bit odd. I didn't touch the LICENSE template or file manually.

File: test/perf.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+describe('skatejs-react-integration', () => {
+
```
Comment (nadiam84) [inline test/perf.js]: I love this huge test :)

**Comment by lukebatchelor** [comment]
LGTM 👍

File: test/perf.js (hunk 2 above)
Comment (treshugart) [inline test/perf.js]: Thanks, wrote it myself ;)

File: lib/index.js [hunk 3]
```diff
@@ -1,111 +0,0 @@
...
-      componentDidMount: function componentDidMount() {
-        var props = this.props;
-
-        // The real custom element is the component that we want to contain the
-        // new render tree as its content.
-        this._realCustomElement = CustomElement(makeCustomElementProps(props));
-
-        // Listen for custom element changes and cancel them to make a stateless component
-        this._realCustomElement.addEventListener('property-change', function (e) {
-          if (props.handlePropertyChange) {
```
Comment (jpnelson) [inline lib/index.js]: There was some behaviour before where if you passed a `handlePropertyChange` prop, it'd cancel all state change events and give that data back to React. Is there an equivalent now?

Comment (treshugart) [inline lib/index.js]: I ended up keeping it and writing tests for it. In doing so, I realised all it's really doing is automating the binding to the `prop-change` event and calling `e.preventDefault()`. Now that we support custom events, all you have to do is simply bind to the event and call `e.preventDefault()` yourself. This prevents us from having to maintain and test this behaviour, and we'd also want to allow customisation of the event name and prop handler name. Now all you need to do is:

```js
<Component onPropChange={e => e.preventDefault()} />
```

...if `propchange` was emitted when a property changes.

**Comment by jpnelson** [comment]
LGTM. This should also fix #11

**Comment by treshugart** [comment]
I'll update #11 to remind us to write a test for it.

**Pull request merged**